src/prog_models/linear_model.py

```diff
@@ -73,13 +73,13 @@ def A(self):
 
     @property
     def B(self):
-        n_inputs = len(self.inputs)
-        n_states = len(self.states)
+        n_inputs = self.n_inputs
```

Comment:
I don't think the declaration adds anything anymore. Let's combine it into line 78, so 78 becomes:
```python
return np.zeros((self.n_states, self.n_inputs))
```

Let's make the same change for the other ones too.